core/src/labels/labels.cpp

```diff
@@ -169,187 +174,135 @@ void Labels::skipTransitions(const std::vector<std::unique_ptr<Style>>& _styles,
     }
 }
 
-void Labels::checkRepeatGroups(std::vector<Label*>& _visibleSet) const {
+bool Labels::labelComparator(const LabelEntry& _a, const LabelEntry& _b) {
```

Comment:
As the discussion we had in the meeting, we could use the depth (ground distance from camera, or perspective factor already available when projecting to screen) to sort  labels within the same priority level. 
Probably to consider out of this PR.